### PR TITLE
[2/3] Add support for multiple exchanges with ccxt (objectified version)

### PR DESCRIPTION
**Pull request by enenn**
## Summary
Adds support for multiple exchanges using the [ccxt library](https://github.com/ccxt/ccxt), and integrates with the new objectified structure. A previous PR implementing ccxt for the old procedural version can be found here: https://github.com/gcarq/freqtrade/pull/503

ccxt is a multi-exchange trading library that supports many different exchanges though a unified API, including Binance and Bittrex. 

**This is PR 2 out of 3 in adding support for ccxt**
- PR 1 (https://github.com/gcarq/freqtrade/pull/585) is mostly implementing ccxt based functions to ` exchange/__init__.py`, and writing tests that validates this part.
- PR 2 is switching the pair format from BTC_XXX to XXX/BTX, and switching ticker interval format from integer minutes (1, 5, 60) to a string (1m, 5m, 1h)
- PR 3 is changing the rest of the bot to use the data structures provided by ccxt when making orders and getting market, history data, and so on. And also various small fixes to make sure tests pass.

## Feedback

**Tests do not pass in this PR and the bot does not run.** That is fixed in the following PRs. Feedback for this specific part of the overarching project to add ccxt is appreciated. 

Testing is restricted to Bittrex and Binance to start with, to keep the testing focused to only a few exchanges.

## Changelog in this PR

This PR is mostly a large find-and-replace job

- Change ticker_interval from integer minutes to string (1,5, 60, 1440) -> (1m, 5m, 1h, 1d)
- Change pair format from 'BTC_XXX' to 'XXX/BTC'
- Change backtesting filename from 'BTC_XXX-5.json' to 'XXX_BTC-5m.json'
- Update backtesting data from Bittrex format to ccxt format

### CODE REVIEW

File: freqtrade/tests/rpc/test_rpc.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def test_rpc_trade_status(default_conf, ticker, mocker) -> None:
     result_message = [
         '*Trade ID:* `1`\n'
         '*Current Pair:* '
-        '[ETH/BTC](https://bittrex.com/Market/Index?MarketName=BTC-ETH)\n'
+        '[BTC_ETH](https://www.bittrex.com/Market/Index?MarketName=BTC-ETH)\n'
```
Comment (xmatthias) [inline freqtrade/tests/rpc/test_rpc.py]: I think this should be `BTC/ETH` - underline as separator should only be used for filenames (in my opinion, anyway). The test is not passing (which is expected ...) - but if this really is the output then it should be changed.

Comment (enenn) [reply freqtrade/tests/rpc/test_rpc.py]: Yeah, messed up while merging. It should not change from the base branch. Thanks!

File: freqtrade/tests/testdata/download_backtest_data.py [hunk 2]
```diff
@@ -10,14 +10,14 @@
...
-        default=None
+    '-p', '--pair',
+    help='JSON file containing pairs to download',
+    dest='pair',
+    default=None
 )
 args = parser.parse_args(sys.argv[1:])
 
-TICKER_INTERVALS = [1, 5]  # ticker interval in minutes (currently implemented: 1 and 5)
+TICKER_INTERVALS = ['1m', '5m']  # ticker interval in minutes (currently implemented: 1 and 5)
```
Comment (gcarq) [inline freqtrade/tests/testdata/download_backtest_data.py]: comment is no longer valid

**Comment by xmatthias** [comment]
can you include the change in the conversion-script to string intervals (5m/1h/...), too? 

https://github.com/gcarq/freqtrade/blob/feat/objectify-ccxt/scripts/convert_backtestdata.py#L139-L141 

this way the conversation-script should work including renaming (i intentionally did not implement that in the script, as it only makes sense with this PR).

File: freqtrade/tests/testdata/download_backtest_data.py (hunk 2 above)
Comment (enenn) [reply freqtrade/tests/testdata/download_backtest_data.py]: Fixed https://github.com/gcarq/freqtrade/pull/600/commits/65c5a0b308490fae1dda5217cf822b51415e0de1

**Comment by xmatthias** [comment]
good catch on the integer overflow.
Honestly, this looks good to me - and i'm already curious for PR3 ... :)

**Pull request merged**